Source/Laser/LaserProfilesImpl/LaserProfileFromTXYEFile.cpp

```diff
@@ -285,32 +261,28 @@ WarpXLaserProfiles::FromTXYEFileLaserProfile::read_data_t_chuck(int t_begin, int
     //Indices of the first and last timestep to read
     auto i_first = max(0, t_begin);
     auto i_last = min(t_end-1, m_params.nt-1);
-    if(i_last-i_first+1 > static_cast<int>(m_params.E_data.size()))
+    if((i_last-i_first+1)*m_params.nx*m_params.ny > static_cast<int>(m_params.E_data.size()))
         Abort("Data chunk to read from file is too large");
 
-    Vector<Real> h_E_data(m_params.E_data.size());
+    Vector<Complex> h_E_data(m_params.E_data.size());
 
     if(ParallelDescriptor::IOProcessor()){
-        //Read data chunk
-        std::ifstream inp(m_params.txye_file_name, std::ios::binary);
-        if(!inp) Abort("Failed to open txye file");
-        inp.exceptions(std::ios_base::failbit | std::ios_base::badbit);
-
-        auto skip_amount = 1 +
-            3*sizeof(uint32_t) +
-            m_params.t_coords.size()*sizeof(double) +
-            m_params.h_x_coords.size()*sizeof(double) +
-            m_params.h_y_coords.size()*sizeof(double) +
-            sizeof(double)*t_begin*m_params.nx*m_params.ny;
-        inp.seekg(skip_amount);
-        if(!inp) Abort("Failed to read field data from txye file");
-        const int read_size = (i_last - i_first + 1)*
-            m_params.nx*m_params.ny;
-        Vector<double> buf_e(read_size);
-        inp.read(reinterpret_cast<char*>(buf_e.dataPtr()), read_size*sizeof(double));
-        if(!inp) Abort("Failed to read field data from txye file");
-        std::transform(buf_e.begin(), buf_e.end(), h_E_data.begin(),
-            [](auto x) {return static_cast<amrex::Real>(x);} );
+
+        auto series = io::Series(m_params.txye_file_name, io::Access::READ_ONLY);
+        auto i = series.iterations[0];
+        auto E = i.meshes["laserEnvelope"];
+        auto E_laser = E[io::RecordComponent::SCALAR];
+
+        // alternatively, pass pre-allocated
+        std::shared_ptr< std::complex<double> > x_data = E_laser.loadChunk< std::complex<double> >();
```

Comment:
For chunking:
```
std::shared_ptr< double > x_data =
    E_x.loadChunk< double >(
        io::Offset{..., ..., ...}, extent);
```
see https://openpmd-api.readthedocs.io/en/0.15.1/usage/firstread.html#register-chunk